overlord/hookstate/ctlcmd/system_mode.go

```diff
@@ -0,0 +1,94 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/devicestate"
+	"github.com/snapcore/snapd/strutil"
+	"gopkg.in/yaml.v2"
+)
+
+type systemModeCommand struct {
+	baseCommand
+}
+
+var shortSystemModeHelp = i18n.G("Get the current system mode information")
+
+var longSystemModeHelp = i18n.G(`
+The system-mode command returns information about the system mode the device is in.
+
+This information includes the mode itself, whether the model snaps have been installed from the seed (seed-loaded). The system mode will be one of run, recover, or install.
+
+The information can also include (factory: true) whether the device booted an image flagged as for factory use ("factory mode"). This can be set for convenience when building the image but no security sensitive decisions should be based on this bit alone.
```

Comment:
I'm not sure whether `factory mode` is a potential return property of `factory` (alongside _true_). I've assumed not and  tweaked the formatting slightly to make it easier to read: `Retrieved information can also include factory-related details: (factory: true) declares whether the device booted an image flagged for factory use and ("factory mode") is set for convenience when building the image. No security sensitive decisions should be based on this bit alone.`

Reply:
sorry, "factory mode" is just an unofficial name, as it's not really a mode, is just install mode as assuming a factory context. I tweaked what you proposed this way:

```
Retrieved information can also include "factory mode" details: 'factory: true' declares whether the device booted an image flagged as for factory use. This flag can be set for convenience when building the image. No security sensitive decisions should be based on this bit alone.
```